tests/test_synced_collection.py

```diff
@@ -520,22 +546,30 @@ def test_update_recursive(self, synced_list):
         self.store(data1)
         assert synced_list == data1
 
-        # inavlid data in file
+        # invalid data in file
         data2 = {'a': 1}
         self.store(data2)
         with pytest.raises(ValueError):
             synced_list.load()
 
+        # reset the file
+        self.store([])
+
+    @given(JSONDataStrategy)
+    @example(None)
     def test_reopen(self, synced_list, testdata):
-        synced_list2 = deepcopy(synced_list)
-        synced_list.append(testdata)
-        synced_list.sync()
-        del synced_list  # possibly unsafe
-        synced_list2.load()
-        assert len(synced_list2) == 1
+        synced_list.clear()
+        synced_list2 = deepcopy(synced_list)  # possibly unsafe
```

Comment:
Why is this possibly unsafe? The previous code did not indicate that it could be unsafe.

Reply:
This is in reference to the [warning](https://github.com/vishav1771/signac/blob/cc895ea425f404125eb50028f3024dfbd604ec78/signac/core/jsoncollection.py#L85-L91) we added to `JSONDict`. I forgot to add this last time.